### DISCUSSION TITLE
Blocked by CORS on Safari : Cannot use wildcard in Access-Control-Allow-Origin when credentials flag is true.

### DISCUSSION THREAD

**Discussion by poof86** [esm-dev/esm.sh#842] [General]
Loading the packages works fine on Chrome but in Safari i'm getting these errors:

<img width="699" alt="Screenshot 2024-06-08 at 15 25 47" src="https://github.com/esm-dev/esm.sh/assets/731498/ade15547-beb7-40a8-a9a3-29c56a15d05a">

Is it something I'm doing wrong or has the CORS on esm.sh changed?
Do I need to use a proxy solution if I want it to work for Safari and it's stricter CORS?

Thanks!

**Comment by ije** [comment]
seems it works for me. can you please share your code?
read this may can help: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSNotSupportingCredentials

**Reply by poof86** [reply]
I'm using an importmap and have my modules loaded in the header like this:
```
<script type="importmap" id="wp-importmap">
{"imports":{"@floating-ui\/dom":"https:\/\/esm.sh\/@floating-ui\/dom?alias=react:preact\/compat,react-dom:preact\/compat,react\/jsx-runtime:preact\/jsx-runtime,@types\/react:preact\/compat\u0026deps=preact@10.19.3","fuse.js":"https:\/\/esm.sh\/fuse.js"}}
</script>
<script type="module" src="https://domain.com/js/view.js?ver=c8b1f7a1c16e1b41" id="view-script-module-js-module" crossorigin="anonymous"></script>

```
I've added the crossorigin attribute, but that didn't seem to help...

**Reply by ije** [reply]
can you please check whether safari tries to send cookies to esm.sh?

**Reply by poof86** [reply]
In the Safari Inspector Network tab I see this for the request:

Request Cookies
No request cookies.

Response Cookies
No response cookies.

Also, when I enable in Safari -> Settings... > Developer Tab > "Disable cross-origin restrictions" all works fine like in Chrome

So some of the requests get a redirect and Safari is not happy with that

Failed to load resource: Cross-origin redirection to https://esm.sh/decimal.js@10.4.3?exports=Decimal denied by Cross-Origin Resource Sharing policy: Cannot use wildcard in Access-Control-Allow-Origin when credentials flag is true.

Is there a way to disable crossorigin credentials for all import requests that are done through an importmap?

**Reply by poof86** [reply]
I'm trying out self-hosting esm.sh to come with a solution.

In my NGINX reverse proxy to the esm.sh docker I'm replacing the 'Access-Control-Allow-Origin' header to not use a wildcard "*" value but use the specific $http_origin with a map of origins I will allow. It seems to work for now, luckily esm.sh has this self hosting feature!
